test/wallet-rpc-test.js

```diff
@@ -0,0 +1,287 @@
+      const receive = 'mkpZhYtJu2r87Js3pDiWJDmPte2NRZ8bJV';
+
+      {
+        await wclient.execute('selectwallet', [standardWalletId]);
+        const response = await wclient.execute('getaddressinfo', [receive]);
+        assert.equal(response.iswatchonly, false);
+      }
+      {
+        await wclient.execute('selectwallet', [watchOnlyWalletId]);
+        const response = await wclient.execute('getaddressinfo', [receive]);
+        assert.equal(response.iswatchonly, true);
+      }
+    });
+
+    // Assert that getaddressinfo returns the same address.
+    // The gotcha would be returning an address with the incorrect
+    // hrp which would be a problem with the network the address
+    // object uses
+    it('should return the correct address', async () => {
+      // m/44'/1'/0'/0/0
+      const receive = 'mkpZhYtJu2r87Js3pDiWJDmPte2NRZ8bJV';
+
+      await wclient.execute('selectwallet', [watchOnlyWalletId]);
+      const response = await wclient.execute('getaddressinfo', [receive]);
+      assert.equal(response.address, receive);
+    });
+
+    // Select the watch only wallet that was created with the xpub
+    // at account index 0', derive the same address locally
+    // and pass to getaddressinfo, assert that ismine is true,
+    // select the standard wallet that has a different hd seed
+    // and then pass the same address to getaddressinfo and
+    // assert that ismine is false
+    it('should detect owned address', async () => {
+      // m/44'/1'/0'/0/0
+      const receive = 'mkpZhYtJu2r87Js3pDiWJDmPte2NRZ8bJV';
+      {
+        await wclient.execute('selectwallet', [watchOnlyWalletId]);
+        const response = await wclient.execute('getaddressinfo', [receive]);
+        assert.equal(response.ismine, true);
+      }
+      {
+        await wclient.execute('selectwallet', [standardWalletId]);
+        const response = await wclient.execute('getaddressinfo', [receive]);
+        assert.equal(response.ismine, false);
+      }
+    });
+
+    it('should detect a p2sh address', async () => {
+      const script = Script.fromMultisig(2, 2, pubkeys);
+
+      const address = Address.fromScript(script);
+      const addr = address.toString(network);
+      const response = await wclient.execute('getaddressinfo', [addr]);
+      assert.equal(response.isscript, true);
+      assert.equal(response.iswitness, false);
+      assert.equal(response.witness_program, undefined);
+    });
+
+    // assert that the correct witness program is returned for a
+    // p2wpkh address
+    it('should return the correct program', async () => {
+      // m/44'/1'/0'/0/5
+      const receive = 'bcrt1q53724q6cywuzsvq5e3nvdeuwrepu69jsc6ulmx';
+      const addr = Address.fromString(receive);
+
+      await wclient.execute('selectwallet', [watchOnlyWalletId]);
+      const str = addr.toString(network);
+      const response = await wclient.execute('getaddressinfo', [str]);
+      assert.equal(response.witness_program, addr.hash.toString('hex'));
```

Comment:
Assert `iswitness: true` here, though that is pretty much handled by witness_program check.